gel/asyncio_client.py

```diff
@@ -346,6 +349,123 @@ async def __anext__(self):
         return iteration
 
 
+class AsyncIOBatchIteration(transaction.BaseTransaction):
+
+    __slots__ = ("_managed", "_locked", "_batched_ops")
+
+    def __init__(self, retry, client, iteration):
+        super().__init__(retry, client, iteration)
+        self._managed = False
+        self._locked = False
+        self._batched_ops = []
+
+    async def __aenter__(self):
+        if self._managed:
+            raise errors.InterfaceError(
+                'cannot enter context: already in an `async with` block')
+        self._managed = True
+        return self
+
+    async def __aexit__(self, extype, ex, tb):
+        with self._exclusive():
```

Comment:
exit should call `await self.wait()` if there are some batched but never waited for operations, right?

Reply:
oh yes, good catch!

Reply:
please add/generate some tests!